### PR TITLE
Fix: Column position in space-infix-ops rule (fixes #2354)

### LINKED ISSUE #2354: Incorrect position reported by space-infix-ops
space-infix-ops always reports the location based on the very first item in boolean expression.

```js
/* eslint space-infix-ops: 2, no-alert: 0 */
/* eslint-env browser */

var test, test2;

var output = test || (test && test.value) ||(test2 && test2.value);

alert(output);
```

reports the following:
```
6:13 - Infix operators must be spaced. (space-infix-ops)
````

I think what should be reported is 6:42 as there are multiple infix operators on this line.

However, when provided with the following code:

```js
/* eslint space-infix-ops: 2, no-alert: 0 */
/* eslint-env browser */

alert(4 + 5*2);
```

it correctly reports:
```
4:10 - Infix operators must be spaced. (space-infix-ops)
```

Tests where done in online demo, so latest version of eslint.

### PR DESCRIPTION
**Pull request by gyandeeps**

### CODE REVIEW

File: lib/rules/space-infix-ops.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ module.exports = function(context) {
         "?", ":", ","
     ];
 
-    function isSpaced(left, right) {
+    function firstNonSpacedNode(left, right) {
```
Comment (nzakas) [inline lib/rules/space-infix-ops.js]: `getFirstNonSpacedNode()` better describes what this is doing.

Comment (nzakas) [inline lib/rules/space-infix-ops.js]: Actually, `getFirstNonSpacedToken()`, since it's returning a token.

File: lib/rules/space-infix-ops.js [hunk 2]
```diff
@@ -27,33 +27,50 @@ module.exports = function(context) {
...
-        return true;
+        return null;
     }
 
-    function report(node) {
-        context.report(node, "Infix operators must be spaced.");
+    function report(mainNode, culpritNode) {
+        var location = {
+            line: culpritNode.loc.start.line,
+            column: culpritNode.range[0]
```
Comment (nzakas) [inline lib/rules/space-infix-ops.js]: This value will be incorrect. It should be `culpritNode.loc.start.col`.

File: lib/rules/space-infix-ops.js [hunk 3]
```diff
@@ -27,33 +27,50 @@ module.exports = function(context) {
...
+        };
+
+        context.report(mainNode, location, "Infix operators must be spaced.");
```
Comment (nzakas) [inline lib/rules/space-infix-ops.js]: Actually, you can just pass `culpritNode.loc.start` as the second argument here.

File: lib/rules/space-infix-ops.js (hunk 1 above)
Comment (gyandeeps) [inline lib/rules/space-infix-ops.js]: Dang it. I had that in my mind but forgot to put it.
Will make the change. Thanks

File: lib/rules/space-infix-ops.js [hunk 4]
```diff
@@ -27,33 +27,50 @@ module.exports = function(context) {
...
+                return op;
             }
         }
-        return true;
+        return null;
     }
 
-    function report(node) {
-        context.report(node, "Infix operators must be spaced.");
+    function report(mainNode, culpritNode) {
```
Comment (nzakas) [inline lib/rules/space-infix-ops.js]: Add some JSDoc here. Is `culpritNode` a node or a token? I think a token, probably should be `culpritToken`.

File: lib/rules/space-infix-ops.js (hunk 3 above)
Comment (gyandeeps) [inline lib/rules/space-infix-ops.js]: Didn't know about this. :)
Will make the change.

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**